### PR TITLE
Enable task relocation for containers in  Eureka/OUT_OF_SERVICE state

### PR DESCRIPTION
**Pull request by tbak**
If a container is in OUT_OF_SERVICE state in Eureka, its real state is lost. This forces us two chose between two strategies:
* assume containers are down, and report them as unhealthy
* assume containers are up, and report them as healthy
The former strategy was assumed before this change. This resulted in large pools of resources to be excluded from the relocation process. We switch to the latter strategy, accepting that this may increase the risk for service availability during rollbacks.

### CODE REVIEW

File: titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java [hunk 1]
```diff
@@ -115,23 +119,28 @@ private ContainerHealthStatus buildHealthStatus(Task task, ContainerHealthState
...
+    private ContainerHealthState takeStateOf(Job<?> job, Task task) {
         // If it is finished, ignore Eureka status
         if (task.getStatus().getState() == TaskState.Finished) {
             return ContainerHealthState.Terminated;
         }
 
+        List<InstanceInfo> instances = eurekaClient.getInstancesById(task.getId());
+
+        // If a job is disabled, the real Eureka state is hidden. If the container is not registered with Eureka in
+        // the disabled job, we also do not put any constrains here. In both cases we report it is healthy.
```
Comment (corindwyer) [inline titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java]: constrains -> constraints

File: titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java [hunk 2]
```diff
@@ -194,13 +216,32 @@ private String takeStateReasonOf(Task task) {
...
     }
 
-    private Optional<ContainerHealthEvent> handleTaskStateUpdate(Task task, ConcurrentMap<String, ContainerHealthEvent> state) {
+    private Flux<ContainerHealthEvent> handleJobEnabledStatusUpdate(Job current, Job previous, ConcurrentMap<String, ContainerHealthEvent> state) {
+
+        if (!JobFunctions.isServiceJob(current)) {
+            return Flux.empty();
+        }
+
+        // We need to examine in if a job 'enabled' status was changed.
```
Comment (corindwyer) [inline titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java]: `We need to examine in if a job 'enabled' status was changed.` -> `Examine if a job's 'enabled' status was changed.`

File: titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java (hunk 1 above)
Comment (tbak) [reply titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java]: Fixed

File: titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java (hunk 2 above)
Comment (tbak) [reply titus-ext/eureka/src/main/java/com/netflix/titus/ext/eureka/containerhealth/EurekaContainerHealthService.java]: Fixed

**Pull request merged**